src/ArticleView.js

```diff
@@ -1,100 +1,51 @@
-  constructor(props) {
-    super(props);
-    this.state = {
-      loaded: false,
-      article: null,
-    };
-  }
-
-  fetchArticle(slug) {
-    client.items()
-      .type('article')
-      .equalsFilter('elements.url_pattern', slug)
+import React, {useState, useEffect} from "react";
+import {Link} from "react-router-dom";
+import {client} from "./config";
+import {resolveContentLink} from "./linkResolver";
+import {resolveItemInRichText} from "./itemResolver";
+
+function ArticleView({match}) {
+  const [isLoading, setLoading] = useState(true);
+  const [article, setArticle] = useState({});
+
+  const fetchArticle = (slug) => {
+    return client
+      .items()
+      .type("article")
+      .equalsFilter("elements.url_pattern", slug)
       .depthParameter(1)
       .queryConfig({
         urlSlugResolver: resolveContentLink,
-        richTextResolver: resolveItemInRichText,
+        richTextResolver: resolveItemInRichText
       })
       .toObservable()
-      .pipe(takeUntil(unsubscribeSubject))
       .subscribe((response) => {
-        console.log(response.items[0]);
-        this.setState({
-          loaded: true,
-          article: response.items[0]
-        });
+        console.log(response.items);
+        setArticle(response.items[0]);
+        setLoading(false);
       });
-  }
-
-  handleClick(event, richTextElement) {
-    if (event.target.tagName === 'A' && event.target.hasAttribute('data-item-id')) {
-      event.preventDefault();
-
-      const id = event.target.getAttribute('data-item-id');
-      const link = richTextElement.links.find(link => link.linkId === id);
-      const newPath = resolveContentLink(link).url;
-      if (newPath) {
-        this.props.history.push(newPath);
-      }
-    }
-  }
-
-  componentDidMount() {
-    let slug = this.props.match.params.slug;
-    this.fetchArticle(slug);
-  }
-
-  componentDidUpdate(oldprops) {
-    let oldSlug = oldprops.match.params.slug;
-    let newSlug = this.props.match.params.slug;
-    if (oldSlug !== newSlug) {
-      this.fetchArticle(newSlug);
-    }
-  }
-
-  componentWillUnmount() {
-    this.unsubscribe();
-  }
-
-  unsubscribe() {
-    unsubscribeSubject.next();
-    unsubscribeSubject.complete();
-  }
-
-  render = (props) => {
-    if (this.state.loaded) {
-      const article = this.state.article;
-      const title = article.title.value;
-      const bodyCopy = article.body_copy;
-
-      return (
-        <div>
-          <Link to="/">Home</Link>
-          <h1>{title}</h1>
-          <div className="article_body"
-            dangerouslySetInnerHTML={{ __html: bodyCopy.resolveHtml() }}
-            onClick={event => this.handleClick(event, bodyCopy)} />
-        </div>
-      );
-    } else {
-      return (
-        <div>
-          Loading...
-        </div>
-      );
-    }
-  }
+  };
+
+  useEffect(() => {
+    const subscription = fetchArticle(match.params.slug);
+    return () => subscription.unsubscribe();
+  }, []);
```

Comment:
Adding `match.params.slug` to the array on line 32 should prevent this effect from running on every render unless the slug changes.

Reply:
@kstulgys could you please adjust the code? we'll be happy to merge it then :)

Reply:
@petrsvihlik Sorry for late response, I hope this is still relevant, added in the useEffect hook dependency. As mentioned above :)